runner/internal/shim/docker.go

```diff
@@ -317,8 +317,12 @@ func createContainer(ctx context.Context, client docker.APIClient, runnerDir str
 		Cmd:          []string{strings.Join(dockerParams.DockerShellCommands(taskConfig.PublicKeys), " && ")},
 		Entrypoint:   []string{"/bin/sh", "-c"},
 		ExposedPorts: exposePorts(dockerParams.DockerPorts()...),
+		Env: []string{
+			"PJRT_DEVICE=TPU",
```

Comment:
Does it mean we always set PJRT_DEVICE=TPU? So even when not running TPU but CUDA? I think we should set PJRT_DEVICE=TPU when running on TPUs only.

Reply:
@r4victor Should I set it as optional flag in docker run  something similar to privileged_flag below?
`nohup dstack-shim {dev_flag} docker --keep-container {privileged_flag} {pjrt_device}>{DSTACK_WORKING_DIR}/shim.log 2>&1 &`

Reply:
I'd set PJRT_DEVICE in the runner where we execute the job and set other envs:

https://github.com/dstackai/dstack/blob/62d25c3029c191dc2d7c7d7f50e09e413d93c1d0/runner/internal/executor/executor.go#L196-L207

This is good because we avoid introducing another place where we set envs. This would require passing whether tpu is used or not to the runner API but this should not be hard to do.

Setting PJRT_DEVICE via shim arg would probably work as well. Feel free to go this route if it works.

Reply:
@r4victor "This is good because we avoid introducing another place where we set envs." I think is a very valid point. I will set in the runner's jobEnvs